package/ios/Sources/RNCandle.swift

```diff
@@ -192,211 +279,250 @@ final class HybridRNCandle: HybridRNCandleSpec {
-    case .whatsapp:
-      return .whatsapp
-    case .messenger:
-      return .messenger
-    case .linkedin:
-      return .linkedin
-    case .discord:
-      return .discord
-    case .messages:
-      return .messages
-    case .telegram:
-      return .telegram
-    case .reddit:
-      return .reddit
-    case .pinterest:
-      return .pinterest
-    case .newYorkTimes:
-      return .newYorkTimes
-    case .washingtonPost:
-      return .washingtonPost
-    case .wallStreetJournal:
-      return .wallStreetJournal
-    case .cnn:
-      return .cnn
-    case .yahoo:
-      return .yahoo
-    case .fox:
-      return .fox
-    case .perplexity:
-      return .perplexity
-    case .openai:
-      return .openai
-    case .polymarket:
-      return .polymarket
-    case .espn:
-      return .espn
-    case .youtube:
-      return .youtube
-    case .netflix:
-      return .netflix
+extension TradeQuoteRequest {
+  var toGained: Models.TradeAssetQuoteRequest {
+    get throws {
+      if let fiatAssetQuoteRequest = gained.fiatAssetQuoteRequest {
+        return Models.TradeAssetQuoteRequest.FiatAssetQuoteRequest(
+          .init(
+            assetKind: .fiat,
+            serviceAccountID: fiatAssetQuoteRequest.serviceAccountID,
+            currencyCode: fiatAssetQuoteRequest.currencyCode,
+            amount: fiatAssetQuoteRequest.amount
+          )
+        )
+      } else if let marketAssetQuoteRequest = gained.marketAssetQuoteRequest {
+        return Models.TradeAssetQuoteRequest.MarketAssetQuoteRequest(
+          .init(
+            assetKind: .init(rawValue: marketAssetQuoteRequest.assetKind) ?? .stock,
+            serviceAccountID: marketAssetQuoteRequest.serviceAccountID,
+            serviceAssetID: marketAssetQuoteRequest.serviceAssetID,
+            symbol: marketAssetQuoteRequest.symbol,
+            amount: marketAssetQuoteRequest.amount
+          )
+        )
+      } else if gained.nothingAssetQuoteRequest != nil {
+        return Models.TradeAssetQuoteRequest.NothingAssetQuoteRequest(
+          .init(assetKind: .nothing)
+        )
+      } else if let transportAssetQuoteRequest = gained.transportAssetQuoteRequest {
+        return Models.TradeAssetQuoteRequest.TransportAssetQuoteRequest(
+          .init(
+            assetKind: .transport,
+            serviceAssetID: transportAssetQuoteRequest.serviceAssetID,
+            originCoordinates: transportAssetQuoteRequest.originCoordinates?.toCoordinates,
+            originAddress: transportAssetQuoteRequest.originAddress?.toAddress,
+            destinationCoordinates: transportAssetQuoteRequest.destinationCoordinates?
+              .toCoordinates,
+            destinationAddress: transportAssetQuoteRequest.destinationAddress?.toAddress,
+            seats: transportAssetQuoteRequest.seats
+          )
+        )
+      } else {
+        throw RNClientError.badInitialization(
+          message: "Unsuppoted TradeQuoteRequest: \(String(describing: self))")
```

Comment:
#req typo, but I'd just change this to `Internal Candle Error: corrupted trade quote request`, same as I suggested on the JS side.